subsystem/elevator.py

```diff
@@ -0,0 +1,204 @@
+                max_length = adjacent / math.cos(angle)
+                # if the length of the angle is greater than the max_elevator_height, return one
+                # else give a percentage of the height of the current line and the max_elevator_height
+                # protects against any irregular numbers heigher than the max elevator height by always
+                # defaulting to 1
+                if abs(max_length) > constants.max_elevator_height:
+                    return 1
+                else:
+                    res = abs(max_length / constants.max_elevator_height)
+                    return res  # where is the hype at
+
+            # Finds the length of each the dimensions for the boundry box using the constants
+            top_vertical_boundry_from_pivot = constants.vertical_boundary - constants.pivot_point_height - constants.top_boundry_buffer_gap
+            bottom_vertical_boundry_from_pivot = constants.pivot_point_height - constants.bottom_boundry_buffer_gap
+            horizontal_boundry_length_from_pivot = 0.5 * (constants.robot_length) + constants.horizontal_boundary - constants.side_boundry_buffer_gap
+            
+            top_boundry_max_angle = math.atan2(top_vertical_boundry_from_pivot, horizontal_boundry_length_from_pivot)
+            bottom_boundry_max_angle = math.atan2(bottom_vertical_boundry_from_pivot, horizontal_boundry_length_from_pivot) + math.radians(90)
+            # if the angle is within the minimun range of each of these boundry sides
+
+            # assuming the top is zero...
+            
+            if(angle < top_boundry_max_angle):
+                return get_length_ratio(angle, top_vertical_boundry_from_pivot)
+            elif(angle > top_boundry_max_angle and angle < bottom_boundry_max_angle):
+                return get_length_ratio(angle, horizontal_boundry_length_from_pivot)
+            elif (angle > bottom_boundry_max_angle):
+                return get_length_ratio(angle, bottom_vertical_boundry_from_pivot)
+
+    def set_length(self, length: meters):  # set arm extension
+        length_ratio = self.boundary_box(self.turn_sensor * (2 * math.pi))
+        if length_ratio * constants.max_elevator_height > length:
+            self.motor_extend.set_target_position(
+                length * constants.elevator_extend_gear_ratio)
+            self.elevator_length = self.motor_extend.get_sensor_position() / \
+                constants.elevator_extend_gear_ratio
+            return True
+        else:
+            self.motor_extend.set_target_position(
+                length_ratio * constants.max_elevator_height * constants.elevator_extend_gear_ratio)
+            return length_ratio * constants.max_elevator_height * constants.elevator_extend_gear_ratio
+
+    def shoulder_rotation_limits(self, radians: float):
+        # returns if the angle is within the soft limits
+        if self.rotation_overide:
+            return True
```

Comment:
I like returning early here. The reason I like it is because you can get rid of the `else` since if code got past that early return, we know  `rotation_overide` must be false.

In general, the less indented your code, the lowest complexity and easier to read.